packages/ember-glimmer/tests/integration/components/closure-components-test.js

```diff
@@ -0,0 +1,669 @@
+    });
+
+    this.render(strip`
+      {{component (component "-looked-up" "Hodari" greeting="Hodi")
+                  greeting="Hola"}}`
+    );
+
+    this.assertText('Hola Hodari');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hola Hodari');
+  }
+
+  ['@test updates when component path is bound']() {
+    this.registerComponent('-mandarin', {
+      template: 'ni hao'
+    });
+
+    this.registerComponent('-hindi', {
+      template: 'Namaste'
+    });
+
+    this.render('{{component (component lookupComponent)}}', {
+      lookupComponent: '-mandarin'
+    });
+
+    this.assertText('ni hao');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('ni hao');
+
+    this.runTask(() => this.context.set('lookupComponent', '-hindi'));
+
+    this.assertText('Namaste');
+
+    this.runTask(() => this.context.set('lookupComponent', '-mandarin'));
+
+    this.assertText('ni hao');
+  }
+
+  ['@test updates when curried hash arguments is bound in block form']() {
+    this.registerComponent('-looked-up', {
+      template: '{{greeting}}'
+    });
+
+    this.render(strip`
+      {{#with (hash comp=(component "-looked-up" greeting=greeting)) as |my|}}
+        {{#my.comp}}{{/my.comp}}
+      {{/with}}`);
+
+    this.assertText('');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('');
+
+    this.runTask(() => this.context.set('greeting', 'Hodi'));
+
+    this.assertText('Hodi');
+
+    this.runTask(() => this.context.set('greeting', undefined));
+
+    this.assertText('');
+  }
+
+  ['@test nested components overwrite named positional parameters']() {
+    this.registerComponent('-looked-up', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: ['name', 'age']
+      }),
+      template: '{{name}} {{age}}'
+    });
+
+    this.render('{{component (component (component "-looked-up" "Sergio" 29) "Marvin" 21) "Hodari"}}');
+
+    this.assertText('Hodari 21');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hodari 21');
+  }
+
+  ['@test nested components overwrite hash parameters']() {
+    this.registerComponent('-looked-up', {
+      template: '{{greeting}} {{name}} {{age}}'
+    });
+
+    this.render(strip`
+      {{#with (component "-looked-up" greeting="Hola" name="Dolores" age=33) as |first|}}
+        {{#with (component first greeting="Hej" name="Sigmundur") as |second|}}
+          {{component second greeting=greeting}}
+        {{/with}}
+      {{/with}}`, {
+        greeting: 'Hodi'
+      });
+
+    this.assertText('Hodi Sigmundur 33');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hodi Sigmundur 33');
+
+    this.runTask(() => this.context.set('greeting', 'Kaixo'));
+
+    this.assertText('Kaixo Sigmundur 33');
+
+    this.runTask(() => this.context.set('greeting', 'Hodi'));
+
+    this.assertText('Hodi Sigmundur 33');
+  }
+
+  ['@skip bound outer named parameters get updated in the right scope']() {
```

Comment:
No, because of the issue with updating positional parameters with the `component` helper.